miner/Miner.hs

```diff
@@ -208,107 +190,114 @@ main = do
     opts = info (pClientArgs <**> helper)
         (fullDesc <> progDesc "The Official Chainweb Mining Client")
 
-scheme :: Env -> (TargetBytes -> HeaderBytes -> IO HeaderBytes)
-scheme env = case cmd $ args env of
-    CPU e -> cpu (version $ args env) e (gen env)
-    GPU _ -> gpu
+run :: RIO Env ()
```

Comment:
Effectively `ReaderT` with built-in support for logging. Mostly a drop-in replacement.